misc/output_test.py

```diff
@@ -12,14 +12,50 @@
 import tempfile
 import unittest
 from textwrap import dedent
-from typing import Dict
+from typing import Dict, Optional
+from collections.abc import Callable
```

Comment:
oops; i forgot to remove the import of `Callable`; this will be gone in the next push.

Reply:
Superior style for avoiding this:

```python
import os
import sys
import tempfile
import foobar
import typing as T

def func(args: T.Optional[T.Dict[str, str]]) -> T.Callable[........]:
    pass
```

A commonly used module where the symbols in use are regularly in flux is a strange time to spend most of your time micromanaging which symbols are currently in scope. But everyone does it. I'm always baffled.

The semantic difference between the two is trivial:
- `from x import y` fills up the local namespace and in return accessing the "y" symbol avoids having to do a name lookup inside "x"

The effects on code readability are considerable:
- with `from x import y` you lose the ability to clearly reason at a glance, that many symbols are part of the same logical context

Reply:
done and pushed; thank you! 

(as i mentioned upthread, i have not touched python in years, so i'm grateful for any python help!)

Reply:
Well, like I said an awful lot of people seem to do it so you're not exactly in bad company. :P 

Probably the rationale is that `from` imports contribute less to line max width. That is the usual reason to use `from` imports for meaningful runtime code.

And if the alternative was truly 
```python
import typing 

def func(args: typing.Optional[typing.Dict[str, str]]) -> typing.Callable[........]:
    pass
```

I'd perhaps concede. That's a 7 char cost per symbol and typing symbols always appear on a line where something else is already present. But `T` is quite distinctive and makes sense to reserve for a language level feature in a way that `frobnicator.core.utils` as `U` doesn't really make sense. :shrug: I wish they'd named it "T" all along.